src/components/button-as-link/index.tsx

```diff
@@ -0,0 +1,37 @@
+import * as React from 'react'
+import styled from 'styled-components'
+import { ButtonSize, StyledButtonProps } from '../button'
+import { applyButtonStyles } from '../button/styles'
+
+/**
+ * The component is used where we need to have
+ * a Button which behaves AS a Link
+ */
+
+const StyledButton = styled.a<StyledButtonProps>`
+  ${(p) =>
+    applyButtonStyles({
+      size: p.size,
+      disabled: p.disabled,
+      inverted: p.inverted,
+    })}
+`
+
+export interface ButtonAsLinkProps
+  extends React.AnchorHTMLAttributes<HTMLAnchorElement>,
+    Partial<StyledButtonProps> {
+  children: React.ReactChild
+}
+
+const ButtonAsLink: React.FC<ButtonAsLinkProps> = ({
+  children,
+  size = ButtonSize.Normal,
+  inverted,
+  ...rest
+}: ButtonAsLinkProps) => (
```

Comment:
Preco vsade duplikujete typ props a pridavate do nich `children`?

Reply:
Eslint 🤷‍♀️
![Screenshot 2020-11-02 at 16 35 57](https://user-images.githubusercontent.com/15198653/97886891-7cd30180-1d29-11eb-973a-2d722fa40ce8.png)

Reply:
> Preco vsade duplikujete typ props a pridavate do nich `children`?

Eslint 🤷‍♀️
![Screenshot 2020-11-02 at 16 35 57](https://user-images.githubusercontent.com/15198653/97886891-7cd30180-1d29-11eb-973a-2d722fa40ce8.png)

Reply:
> > Preco vsade duplikujete typ props a pridavate do nich `children`?
> 
> Eslint 🤷‍♀️
> ![Screenshot 2020-11-02 at 16 35 57](https://user-images.githubusercontent.com/15198653/97886891-7cd30180-1d29-11eb-973a-2d722fa40ce8.png)

to je pro `Preco vsade duplikujete typ props`, ale `pridavate do nich `children`` mate pravdu

Reply:
Riesenie by malo byt fixnut eslint a nie takyto workaround. Taktiez `import * as React` nieje spravny. Tento sposob uz bude onedlho deprecovany.

Reply:
> Riesenie by malo byt fixnut eslint a nie takyto workaround. Taktiez `import * as React` nieje spravny. Tento sposob uz bude onedlho deprecovany.

- `Riesenie by malo byt fixnut eslint a nie takyto workaround`: bavily jsme se o tom v #66 ale nasel jsem jiny sposob https://github.com/yannickcr/eslint-plugin-react/issues/2353#issuecomment-674792754. cc @HormCodes  co si myslete?
- `Taktiez `import * as React` nieje spravny. Tento sposob uz bude onedlho deprecovany.`: mate nejaky clanek? nemuzu najit pro to

Reply:
- to je presne riesenie co som isiel navrhnut, moze to byt bud v overrides len specificky na tsx subory alebo globalne vypnut. Pouzivame toto riesenie uz dlho.
- tak zle som si spominal a nebude deprecovany ale napr. aj [tu](https://reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html#removing-unused-react-imports:~:text=*) odporucaju nadalej nepouzivat takymo sposob importovania

Reply:
> * to je presne riesenie co som isiel navrhnut, moze to byt bud v overrides len specificky na tsx subory alebo globalne vypnut. Pouzivame toto riesenie uz dlho.
> * tak zle som si spominal a nebude deprecovany ale napr. aj [tu](https://reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html#removing-unused-react-imports:~:text=*) odporucaju nadalej nepouzivat takyto sposob importovania

- super  👍
- Mně osobně je jedno, jak psát, jen jsem zvědavý